_projects/quickstart_green_mode.md

```diff
@@ -0,0 +1,59 @@
+---
+collaborating_projects:
+  - coala-quickstart
+  - coala
+desc: "Generate project compatible config files."
+difficulty: medium
+initiatives:
+- GSoC
+- RGSoC
+issues:
+markdown: quickstart_green_mode.md
+mentors:
+  - satwikkansal
+  - jayvdb
+name: "Quickstart Green Mode"
+requirements:
+- "The applicant should have at least one patch accepted to coala-quickstart repository."
+- "The participant should have an understanding of coala's structure and config files."
+- "The proposal needs to contain a clear concept."
+tags:
+- coala-quickstart
+- green mode
+---
+
+coala doesn't help the user unless the user is familiar with its settings and bears.
+coala-quickstart is the project to handle this situation by asking the users general questions about their project and producing a config file to lint their code.
+
+Currently coala-quickstart asks the user a lot of questions, and usually generates a `.coafile` which causes coala-ci to generate errors.
+There is a need for production of *green* config files which adapt the bears and their settings according to the project. This mode would run coala many times in order to answer questions itself, altering bear settings each time to find the set of settings which produces the least amount of errors, and where necessary adds ignore for the remaining files.
+
+This will lead to massive adoption of coala across organizations and user projects as this would initially configure the bears to agree with the repo maintainers existing coding style, as that will result in a quickly mergable Pull Request adding the `.coafile`. They can then make changes to the `.coafile` and their coding style when they want to, and when the bears actually support their repositories intricacies.
+
+
+#### Milestones
+
+##### PREPARATION/BONDING
+
+- Design documentation about how to approach the problem.
+- Make and merge a cEP addressing the following points:
+  - How do you plan to scan through the files for possible bear settings.
+  - How do you plan to tackle conflicting bear settings in the whole project.
+  - What are you plans on reaching other organizations for adoption of coala and the green mode config file in their codebase.
+- The participant should get familiar with the coala-quickstart repository and make plans to utilize the already existing resources provided by coala-quickstart on collecting metadata and detecting inconsistencies in the code.
+
+##### CODING PHASE 1
+
+- Framework for iterating through files has been constructed.
+- Able to detect bear settings based upon the code provided.
+
+##### CODING PHASE 2
+
+- The project delivers one or more config files based on the green mode.
```

Comment:
No.  must be green from coding phase 1.
In coding phase 1, the simple solution is to eliminate bears which produce any errors.
That is a golden rule.  This mode must never produce errors.  This may mean it sometimes spits out a `.coafile` which is empty.

From that solid basis, the coding phases are about finding ways to enable bears with settings which are not the defaults.  ML.